File: workspace/2_SchrijfArabischGetalInRomeinseCijfers/test/num/SchrijfInRomeinseCijfersTest.java
package num;

import static junit.framework.Assert.assertEquals;

import org.junit.Test;

public class SchrijfInRomeinseCijfersTest {
	
	@Test
	public void individueleRomeinseGetallen() {
		assertEquals("I", GetalUtil.alsRomeinsCijfer(1));
		assertEquals("V", GetalUtil.alsRomeinsCijfer(5));
		assertEquals("X", GetalUtil.alsRomeinsCijfer(10));
		assertEquals("L", GetalUtil.alsRomeinsCijfer(50));
		assertEquals("C", GetalUtil.alsRomeinsCijfer(100));
		assertEquals("D", GetalUtil.alsRomeinsCijfer(500));
		assertEquals("M", GetalUtil.alsRomeinsCijfer(1000));
	}
	
	@Test
	public void optellendeRomeinseCijfers() {
		assertEquals("II", GetalUtil.alsRomeinsCijfer(2));
		assertEquals("III", GetalUtil.alsRomeinsCijfer(3));
		assertEquals("VI", GetalUtil.alsRomeinsCijfer(6));
		assertEquals("VII", GetalUtil.alsRomeinsCijfer(7));
		assertEquals("VIII", GetalUtil.alsRomeinsCijfer(8));
		assertEquals("XI", GetalUtil.alsRomeinsCijfer(11));
		assertEquals("XVII", GetalUtil.alsRomeinsCijfer(17));
		assertEquals("XXXVIII", GetalUtil.alsRomeinsCijfer(38));
		assertEquals("LXXII", GetalUtil.alsRomeinsCijfer(72));
		assertEquals("LXXX", GetalUtil.alsRomeinsCijfer(80));
		assertEquals("CXXII", GetalUtil.alsRomeinsCijfer(122));
		assertEquals("CLIII", GetalUtil.alsRomeinsCijfer(153));
		assertEquals("DCLXXVIII", GetalUtil.alsRomeinsCijfer(678));
		assertEquals("MCCXXXIII", GetalUtil.alsRomeinsCijfer(1233));
		assertEquals("MMMCCLXXII", GetalUtil.alsRomeinsCijfer(3272));
	}
	
	@Test
	public void aftrekkendeRomeinseCijfers() {
		assertEquals("IV", GetalUtil.alsRomeinsCijfer(4));
		assertEquals("IX", GetalUtil.alsRomeinsCijfer(9));
		assertEquals("XXXIX", GetalUtil.alsRomeinsCijfer(39));
		assertEquals("XLII", GetalUtil.alsRomeinsCijfer(42));
		assertEquals("XCIX", GetalUtil.alsRomeinsCijfer(99));
		assertEquals("CIX", GetalUtil.alsRomeinsCijfer(109));
		assertEquals("CXIX", GetalUtil.alsRomeinsCijfer(119));
		assertEquals("CXXIX", GetalUtil.alsRomeinsCijfer(129));
		assertEquals("CXXXIX", GetalUtil.alsRomeinsCijfer(139));
		assertEquals("CLXIX", GetalUtil.alsRomeinsCijfer(169));
		assertEquals("CDIII", GetalUtil.alsRomeinsCijfer(403));
		assertEquals("CMLXXIX", GetalUtil.alsRomeinsCijfer(979));
		assertEquals("MCMXCIX", GetalUtil.alsRomeinsCijfer(1999));
	}
}

File: solution/2_SchrijfArabischGetalInRomeinseCijfers/2_SchrijfArabischGetalInRomeinseCijfers/SchrijfInRomeinseCijfersTest.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Linq;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using num;

namespace test
{
    [TestClass]
    public class SchrijfInRomeinseCijfersTest
    {
        [TestMethod]
        public void individueleRomeinseGetallen() {
		    Assert.AreEqual("I", GetalUtil.alsRomeinsCijfer(1));
		    Assert.AreEqual("V", GetalUtil.alsRomeinsCijfer(5));
		    Assert.AreEqual("X", GetalUtil.alsRomeinsCijfer(10));
		    Assert.AreEqual("L", GetalUtil.alsRomeinsCijfer(50));
		    Assert.AreEqual("C", GetalUtil.alsRomeinsCijfer(100));
		    Assert.AreEqual("D", GetalUtil.alsRomeinsCijfer(500));
		    Assert.AreEqual("M", GetalUtil.alsRomeinsCijfer(1000));
	    }
	
	    [TestMethod]
	    public void optellendeRomeinseCijfers() {
		    Assert.AreEqual("II", GetalUtil.alsRomeinsCijfer(2));
		    Assert.AreEqual("III", GetalUtil.alsRomeinsCijfer(3));
		    Assert.AreEqual("VI", GetalUtil.alsRomeinsCijfer(6));
		    Assert.AreEqual("VII", GetalUtil.alsRomeinsCijfer(7));
		    Assert.AreEqual("VIII", GetalUtil.alsRomeinsCijfer(8));
		    Assert.AreEqual("XI", GetalUtil.alsRomeinsCijfer(11));
		    Assert.AreEqual("XVII", GetalUtil.alsRomeinsCijfer(17));
		    Assert.AreEqual("XXXVIII", GetalUtil.alsRomeinsCijfer(38));
		    Assert.AreEqual("LXXII", GetalUtil.alsRomeinsCijfer(72));
		    Assert.AreEqual("LXXX", GetalUtil.alsRomeinsCijfer(80));
		    Assert.AreEqual("CXXII", GetalUtil.alsRomeinsCijfer(122));
		    Assert.AreEqual("CLIII", GetalUtil.alsRomeinsCijfer(153));
		    Assert.AreEqual("DCLXXVIII", GetalUtil.alsRomeinsCijfer(678));
		    Assert.AreEqual("MCCXXXIII", GetalUtil.alsRomeinsCijfer(1233));
		    Assert.AreEqual("MMMCCLXXII", GetalUtil.alsRomeinsCijfer(3272));
	    }

        [TestMethod]
	    public void aftrekkendeRomeinseCijfers() {
		    Assert.AreEqual("IV", GetalUtil.alsRomeinsCijfer(4));
		    Assert.AreEqual("IX", GetalUtil.alsRomeinsCijfer(9));
		    Assert.AreEqual("XXXIX", GetalUtil.alsRomeinsCijfer(39));
		    Assert.AreEqual("XLII", GetalUtil.alsRomeinsCijfer(42));
		    Assert.AreEqual("XCIX", GetalUtil.alsRomeinsCijfer(99));
		    Assert.AreEqual("CIX", GetalUtil.alsRomeinsCijfer(109));
		    Assert.AreEqual("CXIX", GetalUtil.alsRomeinsCijfer(119));
		    Assert.AreEqual("CXXIX", GetalUtil.alsRomeinsCijfer(129));
		    Assert.AreEqual("CXXXIX", GetalUtil.alsRomeinsCijfer(139));
		    Assert.AreEqual("CLXIX", GetalUtil.alsRomeinsCijfer(169));
		    Assert.AreEqual("CDIII", GetalUtil.alsRomeinsCijfer(403));
		    Assert.AreEqual("CMLXXIX", GetalUtil.alsRomeinsCijfer(979));
		    Assert.AreEqual("MCMXCIX", GetalUtil.alsRomeinsCijfer(1999));
	    }
    }
}
